### DISCUSSION TITLE
PYBv1.1 board not detected

### DISCUSSION THREAD

**Discussion by HMC123-puxel** [pyControl/code#128] [General]
Hi, 
I connected the PyControl breakout board (1.2 version) to my PC (Windows 10) using the USB A to micro B cable and a 12V 0.42A 5W power supply. At first, the PC and the GUI detected it (under COM4) but there was an output asking to upgrade the firmware to the one the GUI is using. the instructions required to use DfuSe Demo software but it is not supported anymore and not recommended for use by the developers (I still tried but DruSe demo did not recognized it). instead, they suggest [STM32 Cube Program](https://www.st.com/en/development-tools/stm32cubeprog.html). after trying to use it with under DFU mode with the updated DFU file [first one here - https://micropython.org/resources/firmware/PYBV11-20241129-v1.24.1.dfu] I pressed the reset button (the large one) and since then the blue and red light are constantly on and the board is no longer detected. I also think that the installation of the upgrade was not successful because I do not know the STM32 software and was not sure how to verify the installation of the updated DFU file. Do you have any suggestions? even if I will order a new breakout board, I am not sure how to upgrade the firmware correctly. After that I pressed all push buttons (Reset and USR big and small ones) and connected to different computers with different cables, verified that the system's language did not change...still, not detected. Anything I can do to solve this? or any suggestion how to upgrade for the future?

**Comment by ThomasAkam** [comment]
Hi @HMC123-puxel ,

Have you tried putting the board in dfu mode by physically connecting the dfu pin to the 3.3v pin (see instructions here: https://github.com/micropython/micropython/wiki/Pyboard-Firmware-Update). This may enable you recover from a failed firmware update.  I have only ever updated firmware using DfuSe as detailed here: http://micropython.org/resources/Micro-Python-Windows-setup.pdf . There are some other options listed in the first link above. 

Also, I think the message you saw probably said that the *pyControl framework* version on the pyboard did not match the GUI version. To fix this you just need to reload the pyControl framework to the pyboard using the board config menu in the pyControl GUI, rather than updating the MicroPython version using e.g. DfuSe.

**Reply by HMC123-puxel** [reply]
Hi Thomas, 

Yes, this is exactly what I did and it worked (both the DFU mode so the GUI will recognize again the pyboard) and the reload of the framework. 

Thanks!

**Reply by ThomasAkam** [reply]
Glad to hear the problem is resolved.
